[CI] EsqlClientYamlIT test {p0=esql/160_union_types/suggested_type} failing

**Build Scans:**
- [elasticsearch-periodic #11889 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/wahkqijd3n4u6)
- [elasticsearch-periodic-platform-support #11862 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/yg6zhbm6hfo4g)
- [elasticsearch-periodic-platform-support #11858 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/ynlbjoqdy64ps)
- [elasticsearch-intake #34555 / part3](https://gradle-enterprise.elastic.co/s/wawh6bznycllc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:yamlRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlClientYamlIT.test {p0=esql/160_union_types/suggested_type}" -Dtests.seed=42CA4D6717AF9582 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=pt-Latn-BR -Dtests.timezone=Asia/Bahrain -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
**No**

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlClientYamlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Desql%2F160_union_types%2Fsuggested_type%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: got unexpected warning header [
	299 Elasticsearch-9.4.0-b8804a64ba1ab395902e8e62f3b54a8de5784ed7 "Parameter [default_metric] is deprecated and will be removed in a future version"
]

```

**Issue Reasons:**
- [main] 4 failures in test test {p0=esql/160_union_types/suggested_type} (0.9% fail rate in 461 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (40.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -91,12 +91,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.common.spatial.H3SphericalGeometryTests
   method: testIndexPoints
   issue: https://github.com/elastic/elasticsearch/issues/142439
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlClientYamlIT
-  method: test {p0=esql/160_union_types/suggested_type}
-  issue: https://github.com/elastic/elasticsearch/issues/142525
-- class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
-  method: test {p0=esql/160_union_types/suggested_type}
-  issue: https://github.com/elastic/elasticsearch/issues/142525
 - class: org.elasticsearch.xpack.esql.spatial.SpatialPushDownGeoShapeIT
   method: testQuantizedXY
   issue: https://github.com/elastic/elasticsearch/issues/141234

